core/src/main/java/org/apache/druid/guice/GuiceAnnotationIntrospector.java

```diff
@@ -58,9 +58,9 @@ public Object findInjectableValueId(AnnotatedMember m)
       if (m instanceof AnnotatedMethod) {
         throw new IAE("Annotated methods don't work very well yet...");
       }
-      return Key.get(m.getGenericType());
+      return Key.get(m.getRawType());
```

Comment:
``` java
    /**
     * JDK declared generic type of the annotated element; definition
     * of what exactly this means depends on sub-class. Note that such type
     * cannot be reliably resolved without {@link TypeResolutionContext}, and
     * as a result use of this method was deprecated in Jackson 2.7: see
     * {@link #getType} for replacement.
     *
     * @deprecated Since 2.7 should instead use {@link #getType()}. To be removed from 2.9
     */
```

I'm a little confused that the javadoc indicates that `getType` should be used as a replacement, and current `getGenericType` calls `getRawType()` internally.

Reply:
Hmm good catch.  Here's some more confusion for the pile.  `getType` was marked deprecated in 2.10 as well but seems to be un-deprecated in 2.11+

https://github.com/FasterXML/jackson-databind/blob/2.10/src/main/java/com/fasterxml/jackson/databind/introspect/Annotated.java#L47-L49

https://github.com/FasterXML/jackson-databind/blob/2.11/src/main/java/com/fasterxml/jackson/databind/introspect/Annotated.java#L41-L47

I can swap this to `getType()` instead if that is preferred.

Reply:
I didn't see that `getType` was marked deprecated in 2.10 from the link above. Instead `public final JavaType getType(TypeBindings bogus)` was marked deprecated.

IMO, it's better to follow the javadoc the replace the deprecated API.

Reply:
Sorry for the delay on getting back to you. 

Oof yep I misread.  I will update this Pull Request!

Reply:
Seems like the Javadoc recommendation doesn't work the same and `getRawType` is what we want here.

I'm working on getting tests building locally, I was using JDK17 which I see is a no-no in the build instructions.  Sorry for the churn on this.  Are you okay with pulling `[671c984](https://github.com/apache/druid/pull/12373/commits/671c984e9bcee2b1651d5fa96b5c670da5802822)` as is?